registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -242,83 +278,139 @@ NormalDistributionsTransform<PointSource, PointTarget>::computeAngleDerivatives
   Eigen::Matrix<double, 8, 1> point_angular_jacobian = angular_jacobian_ * Eigen::Vector4d(x[0], x[1], x[2], 0.0);
-  point_jacobian_ (1, 3) = point_angular_jacobian[0];
-  point_jacobian_ (2, 3) = point_angular_jacobian[1];
-  point_jacobian_ (0, 4) = point_angular_jacobian[2];
-  point_jacobian_ (1, 4) = point_angular_jacobian[3];
-  point_jacobian_ (2, 4) = point_angular_jacobian[4];
-  point_jacobian_ (0, 5) = point_angular_jacobian[5];
-  point_jacobian_ (1, 5) = point_angular_jacobian[6];
-  point_jacobian_ (2, 5) = point_angular_jacobian[7];
+  point_jacobian (1, 3) = point_angular_jacobian[0];
+  point_jacobian (2, 3) = point_angular_jacobian[1];
+  point_jacobian (0, 4) = point_angular_jacobian[2];
+  point_jacobian (1, 4) = point_angular_jacobian[3];
+  point_jacobian (2, 4) = point_angular_jacobian[4];
+  point_jacobian (0, 5) = point_angular_jacobian[5];
+  point_jacobian (1, 5) = point_angular_jacobian[6];
+  point_jacobian (2, 5) = point_angular_jacobian[7];
 
   if (compute_hessian)
   {
     Eigen::Matrix<double, 15, 1> point_angular_hessian = angular_hessian_ * Eigen::Vector4d(x[0], x[1], x[2], 0.0);
 
     // Vectors from Equation 6.21 [Magnusson 2009]
-    const Eigen::Vector3d a(0, point_angular_hessian[0], point_angular_hessian[1]);
-    const Eigen::Vector3d b(0, point_angular_hessian[2], point_angular_hessian[3]);
-    const Eigen::Vector3d c(0, point_angular_hessian[4], point_angular_hessian[5]);
-    const Eigen::Vector3d d = point_angular_hessian.block<3, 1>(6, 0);
-    const Eigen::Vector3d e = point_angular_hessian.block<3, 1>(9, 0);
-    const Eigen::Vector3d f = point_angular_hessian.block<3, 1>(12, 0);
+    Eigen::Vector4d a(0, point_angular_hessian[0], point_angular_hessian[1], 0.0f);
+    Eigen::Vector4d b(0, point_angular_hessian[2], point_angular_hessian[3], 0.0f);
+    Eigen::Vector4d c(0, point_angular_hessian[4], point_angular_hessian[5], 0.0f);
+    Eigen::Vector4d d(point_angular_hessian[6], point_angular_hessian[7], point_angular_hessian[8], 0.0f);
+    Eigen::Vector4d e(point_angular_hessian[9], point_angular_hessian[10], point_angular_hessian[11], 0.0f);
+    Eigen::Vector4d f(point_angular_hessian[12], point_angular_hessian[13], point_angular_hessian[14], 0.0f);
 
     // Calculate second derivative of Transformation Equation 6.17 w.r.t. transform vector.
     // Derivative w.r.t. ith and jth elements of transform vector corresponds to the 3x1 block matrix starting at (3i,j), Equation 6.20 and 6.21 [Magnusson 2009]
-    point_hessian_.block<3, 1>(9, 3) = a;
-    point_hessian_.block<3, 1>(12, 3) = b;
-    point_hessian_.block<3, 1>(15, 3) = c;
-    point_hessian_.block<3, 1>(9, 4) = b;
-    point_hessian_.block<3, 1>(12, 4) = d;
-    point_hessian_.block<3, 1>(15, 4) = e;
-    point_hessian_.block<3, 1>(9, 5) = c;
-    point_hessian_.block<3, 1>(12, 5) = e;
-    point_hessian_.block<3, 1>(15, 5) = f;
+    point_hessian.block<4, 1>((9/3)*4, 3) = a;
```

Comment:
Any reason for using (9/3)*4? Perhaps, modify the note in the line above to say:
```
... block matrix starting at (4i, j) ...
```
and update the indices to be of the same form, i.e. 4*3, 4*4, 4*5

Reply:
I changed it to ```(3*4, 3)``` and added a note to indicate that it corresponds to the (3, 3) element of the equation in the paper